packages/cascara/src/specs/Table/Table.spec.mdx

```diff
@@ -1,7 +1,155 @@
+
+## Table Locations
+
+### Admin
+
+- BRC (Grid)
+- Storefront Designer (grid)
+- Bundles (grid)
+- Role Bundles (grid)
+- Onboarding Activities/Schedule
+- Onboarding Activities/Performance
+- Service Department
+- Service Teams
+- Barista FAQS
+- Barista FAQS/Topics
+- Users
+- Integrations/CSV/Import History
+- Integrations/CSV/Import Schedule
+- Integrations/Email/Barista Inboxes
+- Reports
+- Locations
+
+### Barista
+
+- None
+
+### Caffeine
+
+- Entities
+- Training
+- Variables
+- Logs (future)
+- Users (future)
+- Configurations (future)
+- Localization(future)
+
+### Status
+
+- Packages
+
+### Doppio
+
+- Applications
+- Applications/[Application]
+- Intents
+- Intents/[Intent]
+- FAQs
+- FAQS/[FAQ]
+- Entities
+- Archetype Conditions
+- Attributes
+- Hooks
+- Synonyms
+- Synonyms/[Synonym]
+- Variables
+- Variables/[Variable]
+- Training
+- Spell Corrector Dictionary
+- Utilities
+- ESP Service Departments
+- Service Team Classifications
+- Task Classifications
+- Service Departments
+- Service Teams
+- Task Categories
+- Testing Reports
+- Test Phrases
+- Test Sets
+- Tasks
+- Tasks/[Task]
+- Categories
+- Categories/[Category]
+- Outages
+- Outages/[Outage]
+- Integrations
+- Integrations/[Integration]
+- Service Requests
+- Service Request/[Service Request]
+- Model Bindings
+- Widgets
+- Users
+- Users/[User]
+- Locations
+- Assets
+- Job Roles
+- Classifiers
+- Data Sources
+- Data Sources/[Data Source]
+- Annotations
+- Annotations/[Annotation]
+- Smart Rules
+- Smart Rules/[Rule]
+- Flagged Tickets
+- Surveys
+- Configurations
+- Attachments
+- Business Events
+- Localization
+
+## Existing Table Functionality
+
+### CRUD Capabilities
+
+- Ability to add a data row
+- Ability to edit a table rown (inline optional)
+- Ability to delete a row (optional)
+
+### Table Header
+
+- Column headers with labels and optional descriptions
+- Sortable column headers
+
+### Pagination
+
+- Ability to show total number of records available
+- Ability to show visible number of records
+- Ability to show how many pages of records are available
+- Ability to navigate to a specific page of results
+- Ability to navigate to next or previous page of results
+
+### Table Row
+
+- Ability to have cells/columns with specific defined data structures, form fields, action elements (buttons, menus)
+- Ability for results view to be accessible via URL so results can be shared
+
+## Needed/Suggested/Future Table Functionality
+
+- Zebra striping on rows
+- Ability to bulk edit, delete, add records
+- Table header - horizontal VS vertical
+- Ability to customize columns visible
+- Ability to reorganize position/order of columns
+- Ability to have header/columns be sticky
+- Accordion columns
+- Nested tables
+- Tables with a tree like structure
+- Way to handle truncated/text that is too long to fit
+- Mobile view
+- Ability to set a default sort on a column
```

Comment:
@CaffeinatedAllie @lordpixel IMO this should actually not be a feature of the table itself. We pass data into the table and then tell the table what to display. So if we want to change that, I think the best way to handle that is to change the array of `data` being passed into the table.  That is going to lead to less confusion I think in terms of how that data displays. If we are changing the order inside of the table itself, that becomes more complex in terms of figuring it out. We can have some utils for handling that external from the table.

@lordpixel that would also mean that we will need to define a way for hooking this functionality up to table headers. When we define dataConfig we may want to specify a `sorting` attribute for each object. Enums could be `asc` or `desc`. These would be for display only so the header could indicate which cell was doing the sorting and in what direction. Each header cell would have a click handler that could fire whatever func was passed in as part of that dataConfig object as well. Essentially the table will then just display what it is told in the header row too. And when clicked, fire whatever associated function is needed. Sorting of data can happen external. And if there is any strange Array.map work we need to do to results externally can also happen outside of Table.

Reply:
-[ ] @lordpixel @CaffeinatedAllie I think we remove this line? Or if we leave it, make a note that the header cells should be clickable and DISPLAY a sort order for a column?

Reply:
@brianespinosa we can do the following:

1. add the sort definition in the column schema and supporting enums
2. expose an event `onSort` or something similar with the sig. (attribute, direction) => {}

If the no sort config is passed to the column, the buttons in the header won't render.

Reply:
@lordpixel I think we could make it even more simple. Having discreet onEVENT functions decorating the Table forces us down specific paths. If we instead give more implementation agnostic patterns to developers, they can potentially do more.

`dataConfig` could have these params included in each object:

```
{
  sorting: 'asc',
  onHeaderClick: handleDataSortAndStuff()
  ...
}
```

The onHeaderClick would be mapped to an onClick function on the header cells. That handler function could do all kinds of things outside of the Table. It could set and remove `sorting` attributes to other objects in `dataConfig` and then change them when the user clicks. It could also change data in `data` being passed to the table. These changes will cause the table to render and update. This can be achieved without any `onSort` function on the Table itself.

From a DX perspective... maybe there will appear some consistent scenarios where we want to expose some `onSOMETHING` functions on these components. I think my worry is that it might still be early to do that in a consistent way. Whatever functions we expose on the tables, I think we want to make sure they are consistent from one component to the next. And if we force ourselves to follow some consistent React composition patterns first, I think we will have fewer bugs and impossible states to fight with.